packages/legend-query/src/stores/QueryBuilderPropertyEditorState.ts

```diff
@@ -185,17 +268,19 @@ const fillDerivedPropertyArguments = (
       ),
       ...propertyArguments,
     ],
-    queryBuilderState.observableContext,
+    derivedPropertyExpressionState.queryBuilderState.observableContext,
   );
 };
 
 export class QueryBuilderDerivedPropertyExpressionState {
   queryBuilderState: QueryBuilderState;
   path: string;
   title: string;
-  readonly derivedProperty: DerivedProperty;
-  readonly propertyExpression!: AbstractPropertyExpression;
-  readonly parameters: VariableExpression[] = [];
+  propertyExpression: AbstractPropertyExpression;
+  derivedProperty: DerivedProperty;
+  parameters: VariableExpression[] = [];
+  processingDate: ValueSpecification | undefined;
```

Comment:
why do we have a state here for these as well? I thought we control them via `QueryBuilderSetupState`